logger/structured_logger.go

```diff
@@ -0,0 +1,161 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//go:build structuredlogging
```

Comment:
may be worth adding a package level [doc comment](https://go.dev/doc/comment) that explain a bit the package and the two build flag modes, simple vs structured